pyrokinetics/pyroscan.py

```diff
@@ -178,17 +205,91 @@ def load_default_parameter_keys(self):
         parameter_location = ['deuterium', 'a_ln']
         self.add_parameter_key(parameter_key, parameter_attr, parameter_location)
 
+    def load_gk_output(self):
+        """
+        Loads GKOutput as a CleverDict
+
+        Returns
+        -------
+        self.gk_output : CleverDict of data
+        self.gk_output.data : xarray DataSet of data
+        """
+
+        import xarray as xr
+        from cleverdict import CleverDict
+
+        # Set up output as CleverDict
+        self.gk_output = CleverDict()
+
+        # xarray DataSet to store data
+        ds = xr.Dataset(self.parameter_dict)
+
+        if not self.base_pyro.numerics.nonlinear:
```

Comment:
It might be worth allowing this method to take a function as an argument -- this can then detail what is loaded, what analysis is done etc. It could default to a method which implements this linear analysis, for example.

Reply:
So something along the lines of?
```python
def load_gk_output(self, output_func, data_name):
     for pyro in all_pyros:
           output_data = output_func(pyro)

     ds['data_name'] = output_data
```
where

```python
def output_func(pyro):
     output_data = get_output_from_pyro
  
     return output_data
```

Reply:
Something like that, although potentially returning more data.

Reply:
I can leave it like this for now and the make a separate issue for this as it would be nice to have this for a general Pyro object rather than just a Pyroscan for now